screens/Home.js

```diff
@@ -5,26 +5,43 @@ import {
   Image,
   ScrollView,
   Platform,
-  Pressable,
+  ActivityIndicator,
 } from "react-native";
-import { Block, Text, theme } from "galio-framework";
-
+import { Block, theme } from "galio-framework";
 import { Input } from "../components";
 import MasonryList from "../components/MasonryList";
 import { HeaderHeight } from "../constants/utils";
-import usables from "../constants/usables";
-
 import { FontAwesome } from "@expo/vector-icons";
-import { useNavigation } from "@react-navigation/native";
 import ImageUploadComponent from "../components/ImageUploadBtn";
 import Product from "../components/Product";
+import { getExploreApiUrl } from "../api/url";
 
-const { width, height } = Dimensions.get("screen");
-
-const thumbMeasure = (width - 48 - 32) / 3;
-const loading = false;
+const { width } = Dimensions.get("screen");
 
 const Home = () => {
+  const [result, setResult] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const url = getExploreApiUrl();
+  const getData = useCallback(async (feedDataURL) => {
+    setLoading(true);
+    try {
+      const response = await fetch(feedDataURL);
+      const searchResultsData = await response.json();
+      setResult(searchResultsData.data);
+    } catch (error) {
+      Alert.alert(
+        "Error",
+        "We're unable to fetch data right now. Please try again later."
+      );
+    } finally {
+      setLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    getData(url);
+  }, [url]);
```

Comment:
we don't need to pass url to the function and in the dependency array. the url is constant